radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -398,6 +390,159 @@ private void storeMetadata(Transaction dbTxn, LedgerAndBFTProof ledgerAndBFTProo
+  }
+
+  private void storeForksVotingResults(
+      Transaction dbTxn, ImmutableSet<ForkVotingResult> forksVotingResults) {
+    forksVotingResults.forEach(
+        forkVotingResult ->
+            forksVotingResultsDatabase.put(
+                dbTxn,
+                new DatabaseEntry(Longs.toByteArray(forkVotingResult.epoch())),
+                new DatabaseEntry(encodeForkVotingResult(forkVotingResult))));
+  }
+
+  @Override
+  public CloseableCursor<ForkVotingResult> forkVotingResultsCursor(
+      long fromEpoch, long toEpoch, HashCode candidateForkId) {
+    final Cursor underlyingCursor = forksVotingResultsDatabase.openCursor(null, null);
+
+    OperationStatus cursorStatus;
+    try {
+      cursorStatus =
+          underlyingCursor.getSearchKeyRange(
+              new DatabaseEntry(Longs.toByteArray(fromEpoch)), null, DEFAULT);
+    } catch (Exception e) {
+      // close the cursor and re-throw
+      underlyingCursor.close();
+      throw e;
+    }
+
+    if (cursorStatus != SUCCESS) {
+      underlyingCursor.close();
+      return CloseableCursor.empty();
+    }
+
+    return new CloseableCursor<>() {
+      private Optional<ForkVotingResult> nextOpt =
+          findNextForkVotingResult(toEpoch, candidateForkId, underlyingCursor, true);
+
+      @Override
+      public boolean hasNext() {
+        return nextOpt.isPresent();
+      }
+
+      @Override
+      public ForkVotingResult next() {
+        final var res = nextOpt.orElseThrow(NoSuchElementException::new);
+        nextOpt = findNextForkVotingResult(toEpoch, candidateForkId, underlyingCursor, false);
+        return res;
+      }
+
+      @Override
+      public void close() {
+        underlyingCursor.close();
+      }
+    };
+  }
+
+  private Optional<ForkVotingResult> findNextForkVotingResult(
+      long toEpoch, HashCode candidateForkId, Cursor cursor, boolean startWithCurrent) {
+    final DatabaseEntry key = new DatabaseEntry();
+    final DatabaseEntry value = new DatabaseEntry();
+    OperationStatus operationStatus =
+        startWithCurrent
+            ? cursor.getCurrent(key, value, DEFAULT)
+            : cursor.getNext(key, value, DEFAULT);
+
+    if (operationStatus != SUCCESS) {
+      return Optional.empty();
+    }
+
+    long epoch = Longs.fromByteArray(key.getData());
+
+    while (operationStatus == SUCCESS && epoch < toEpoch) {
+      final var nextForkVotingResult = decodeForkVotingResult(epoch, value.getData());
+      if (nextForkVotingResult.candidateForkId().equals(candidateForkId)) {
+        return Optional.of(nextForkVotingResult);
+      }
+      operationStatus = cursor.getNext(key, value, DEFAULT);
+      if (operationStatus == SUCCESS) {
+        epoch = Longs.fromByteArray(key.getData());
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  private byte[] encodeForkVotingResult(ForkVotingResult forkVotingResult) {
+    return Bytes.concat(
```

Comment:
What are your thoughts on storing a DSON object instead of a byte encoding? As we'd then be less fragile to any future requirements changes about what exactly we want to store (and how long the various fields are)

Reply:
I think it's worth to optimize for storage size in this particular case